### PR TITLE
Added some beginning of a CLI

### PR DESCRIPTION
**Pull request by PonteIneptique**

### CODE REVIEW

**Comment by Jean-Baptiste-Camps** [comment]
Great to now have the beginning of commands with options to use Pandora. Not many remarks on my part. 
About the install:
Is it necessary to include the use of a virtualenv in the install process ? It might be left to the user to choose it or not.

**Comment by PonteIneptique** [comment]
It's recommendation in the readme. However, readme are only about recommendation :) Users can do what they want. (I would heavily recommend to always use virtual environments)

**Comment by Jean-Baptiste-Camps** [comment]
Ok, you're the Python expert. I just meant it to maintain installation as standard and simple as possible, but no problem. BTW, should it no be git clone (rather than pull) ?

**Comment by PonteIneptique** [comment]
The reason behind virtualenv is reproducability and conflicts : making sure your scripts are based on specific version of libraries which you can list without having any interference from outside is I think a major thing for academia (it is for programming as well but academia even more) :) 
I'll fix the typo tomorrow.

**Comment by mikekestemont** [comment]
I am not in favour of using virtualenv either because there many competing environment solutions that people use. Is it really necessary? About the CUDA: should this not the responsibility of the user to take of care of this part? Now, we have to adapt our repo whenever something external to Pandora changes to the installation process.

**Review by mikekestemont — COMMENTED** [review]
The get_value is not a bug, but only works when you use theano as a backend. Your solution is also correct, but will only work with tensorflow. IMO, the backend-independent solution is:
old_lr  = K.get_value(self.model.optimizer.lr)
new_lr = np.float32(old_lr * 0.5)
K.set_value(self.model.optimizer.lr, new_lr)

File: train.py [hunk 1]
```diff
@@ -2,63 +2,59 @@
...
     print('::: ended :::')
 
+
 if __name__ == '__main__':
-    main()
+    parser = argparse.ArgumentParser(description="Training interface of Pandora")
+    parser.add_argument("config", help="Path to retrieve configuration file")
+    parser.add_argument("--dev", help="Path to retrieve configuration file")
+    parser.add_argument("--train", help="Path to retrieve configuration file")
+    parser.add_argument("--nb_epochs", help="Number of epoch", type=int)
```
Comment (mikekestemont) [inline train.py]: If we we specify this possibility for one hyperparameter, should we not do for every parameter?

**Comment by PonteIneptique** [comment]
I'll try to answer to the many questions : 

>  competing environment solutions that people use. Is it really necessary? 

Keep in mind it is only a readme, in no way it enforces anything. It is not the installation process and people can install it the way they want. That just represents one way to install the dependencies. It is not an automatically run scripts or whatever.

> About the CUDA: should this not the responsibility of the user to take of care of this part?

We are not enforcing installation of CUDA. Rather, we propose to list of requirements, the only difference being the non-CUDA one uses `tensorflow`, the CUDA one `tensorflow-gpu`. 

>  Now, we have to adapt our repo whenever something external to Pandora changes to the installation process.

We do not need to update those requirements lists except if we add a new dependency ourselves. People are responsible for setting up the rest of their environment, we are just giving the list of basic python dependencies we require to run. Nothing more. This is standard practice to give such an install list.

> Backend independent solution

This is an error from me, I merged the wrong part of the conflict.

> If we we specify this possibility for one hyperparameter, should we not do for every parameter?

I added this specific one because right now it allowed me to check that the process was running without running it 10 times. We can expand to many more parameters, but I did not have the exact human translation of each of these :)

**Comment by PonteIneptique** [comment]
> Backend independent solution

I actually did keep your correction in the end, so that's good, I don't have to fix this part. Keep in mind the former commits where worded before we decided to work this thing out together ;) The PR will be squashed to a simple commit

**Comment by mikekestemont** [comment]
> I'll try to answer to the many questions :
>
> competing environment solutions that people use. Is it really necessary?
>
> Keep in mind it is only a readme, in no way it enforces anything. It is
> not the installation process and people can install it the way they want.
> That just represents one way to install the dependencies. It is not an
> automatically run scripts or whatever.
>
Ok, but then as the first option, I would also include an env-free
installatation guidelines for novice users, who might get confused by this.

> Backend independent solution
>
> This is an error from me, I merged the wrong part of the conflict.
>
okido.

> If we we specify this possibility for one hyperparameter, should we not do
> for every parameter?
>
> I added this specific one because right now it allowed me to check that
> the process was running without running it 10 times. We can expand to many
> more parameters, but I did not have the exact human translation of each of
> these :)
>
Ok. Documentation will follow soon. Promise.

**Comment by mikekestemont** [comment]
ok, will merge.

Prof. Dr. Mike Kestemont | www.mike-kestemont.org | Twitter:
@Mike_Kestemont | mike.kestemont@uantwerp.be | mike.kestemont@gmail.com |
University of Antwerp | City Campus, Prinsstraat 13, room D. 118 I B-2000
Antwerp, Belgium | tel. +32 (0)3 265.42.54

> Check out our documentary on Digital Humanities and Hildegard of Bingen:
watch it in HD on Vimeo: https://vimeo.com/70881172

**Comment by PonteIneptique** [comment]
Let merge with the correction ;)

**Pull request merged**